make/autoconf/basic.m4

```diff
@@ -134,17 +134,31 @@ AC_DEFUN_ONCE([BASIC_SETUP_BUILD_ENV],
   )
   AC_SUBST(BUILD_ENV)
 
+  AC_MSG_CHECKING([for locale to use])
   if test "x$LOCALE" != x; then
     # Check if we actually have C.UTF-8; if so, use it
     if $LOCALE -a | $GREP -q -E "^C\.(utf8|UTF-8)$"; then
       LOCALE_USED=C.UTF-8
+      AC_MSG_RESULT([C.UTF-8 (recommended)])
+    elif $LOCALE -a | $GREP -q -E "^en_US\.(utf8|UTF-8)$"; then
+      LOCALE_USED=en_US.UTF-8
+      AC_MSG_RESULT([en_US.UTF-8 (acceptable fallback)])
     else
-      AC_MSG_WARN([C.UTF-8 locale not found, using C locale])
-      LOCALE_USED=C
+      # As a fallback, check if users locale is UTF-8. USER_LOCALE was saved
+      # by the wrapper configure script before autconf messed up LC_ALL.
+      if $ECHO $USER_LOCALE | $GREP -q -E "\.(utf8|UTF-8)$"; then
+        LOCALE_USED=$USER_LOCALE
+        AC_MSG_RESULT([$USER_LOCALE (untested fallback)])
+        AC_MSG_WARN([Could not find C.UTF-8 or en_US.UTF-8 locale. This is not supported, and the build might fail unexpectedly.])
+      else
+        AC_MSG_RESULT([no UTF-8 locale found])
+        AC_MSG_ERROR([No UTF-8 locale found. This is required for building successfully.])
```

Comment:
Seems we run into this else  on  AIX

```
checking for locale to use... no UTF-8 locale found
configure: error: No UTF-8 locale found. This is required for building successfully.
configure exiting with result code 1
```
maybe it would be nice to display the desired ones C.UTF-8 or en_US.UTF-8 in this message too for more clarity?  (have to check if there are other names on AIX)

Reply:
If you have a locale named `<something>.UTF-8` as your active locale, that will also be accepted, so it is not limited to C and en_US. But it might be an idea to include it in the error message, yes.

Reply:
@MBaesken I added an informative text about supported locales. I also lowered the fatal error of missing an UTF-8 locale into just a warning (for all platforms, not just AIX). Apparently it works with building on AIX for now, even with a undefined character encoding, but I will not guarantee that it will continue to work for all eternity. But a warning is probably be enough, and if it does work the user is free to try. :)

Reply:
> @MBaesken I added an informative text about supported locales. I also lowered the fatal error of missing an UTF-8 locale into just a warning (for all platforms, not just AIX). Apparently it works with building on AIX for now, even with a undefined character encoding, but I will not guarantee that it will continue to work for all eternity. But a warning is probably be enough, and if it does work the user is free to try. :)


Sounds good to me ,  thanks for adjusting !